### PR TITLE
add support for Raspberry Pi 4

### PR DESCRIPTION
**Pull request by HiassofT**
port of #3596 with separate raspberrypi-4.19 kernel and without kodi patches

### CODE REVIEW

**Comment by HiassofT** [comment]
RPi 4.19 kernel with RPi4 support is now up on official kernel repo, I've fixed the SHA256 to match that

File: projects/RPi/options [hunk 1]
```diff
@@ -12,6 +12,9 @@
           TARGET_CPU="arm1176jzf-s"
         elif [ "$DEVICE" = "RPi2" -o "$DEVICE" = "Slice3" ]; then
           TARGET_CPU="cortex-a7"
+        elif [ "$DEVICE" = "RPi4" ]; then
+          TARGET_CPU="cortex-a53"
```
Comment (antonlacon) [inline projects/RPi/options]: TARGET_CPU="cortex-a72" ? Will need config/arch.arm updated too.

**Comment by lrusak** [comment]
Please drop the ffmpeg HEVC patches

File: projects/RPi/options (hunk 1 above)
Comment (HiassofT) [reply projects/RPi/options]: I've been testing with cortex-a72 but that lead to gdb spinning forever on a backtrace (eg when crashlog is created or when manually doing "thread apply all bt")
```
      Thread 15 (Thread 0xa51bc370 (LWP 539)):
      #0  futex_abstimed_wait_cancelable (private=0, abstime=0xa51bb930, expected=0,
          futex_word=0x3884df8) at ../sysdeps/unix/sysv/linux/futex-internal.h:205
      #1  __pthread_cond_wait_common (abstime=0xa51bb930, mutex=<optimized out>,
          cond=0x3884dd0) at pthread_cond_wait.c:539
      #2  __pthread_cond_timedwait (cond=0x3884dd0, mutex=0x0, abstime=0xa51bb930)
          at pthread_cond_wait.c:667
      #3  0x008b6130 in ?? ()
      #4  0x008bd58c in ?? ()
      #5  0x008219ac in XBMCAddon::xbmc::Monitor::waitForAbort(double) ()
      #6  0x0084fbac in ?? ()
      #7  0xb6c0a018 in PyEval_EvalFrameEx () from /usr/lib/libpython2.7.so.1.0
      #8  0xb6c0a018 in PyEval_EvalFrameEx () from /usr/lib/libpython2.7.so.1.0
      #9  0xb6c0a018 in PyEval_EvalFrameEx () from /usr/lib/libpython2.7.so.1.0
```

As we didn't have the time to investigate this issue further I went back to cortex-a53 which didn't show this behaviour.

But I agree, we will need to revisit this at some point and if the issues are solved we can switch to cortex-a72

**Comment by HiassofT** [comment]
@lrusak I've dropped all ffmpeg patches from the PR.

This wil build ffmpeg for RPi4 with `--disable-rpi` which I think is the right thing to do as we neither need mmal acceleration nor the RPi0-3 specific HEVC stuff (which won't work on RPi4 anyways).

File: projects/RPi/options (hunk 1 above)
Comment (antonlacon) [reply projects/RPi/options]: TARGET_CPU_FLAGS="+crc" should go in here too.

Comment (HiassofT) [reply projects/RPi/options]: TARGET_CPU_FLAGS="+crc" is already there, in the next line :-)

File: config/graphic [hunk 2]
```diff
@@ -107,7 +107,7 @@ get_graphicdrivers() {
   fi
 
   if listcontains "${GRAPHIC_DRIVERS}" "vc4"; then
-    GALLIUM_DRIVERS+=" vc4"
+    GALLIUM_DRIVERS+=" vc4 v3d kmsro"
```
Comment (antonlacon) [inline config/graphic]: The previous RPi's will presumably shift to vc4 at some point, so this should be behind a if $DEVICE = RPi4?

**Comment by antonlacon** [comment]
Merged into my RPi3 branch and built/ran without issue. Looks ok to me aside from what was noted. Cheers.

File: config/graphic (hunk 2 above)
Comment (HiassofT) [reply config/graphic]: enabling v3d and kmsro drivers won't have a negative impact on older RPis (except for wasting a bit of space) and I'd rather keep PROJECT/DEVICE selection out of common config files and scripts like this, if possible

**Comment by HiassofT** [comment]
I've updated bcmstat and brcmfmac_sdio-firmware-rpi commits to drop the local patches and update to latest versions instead.

Kernel is now at 4.19.56 and firmware at Jun 25 (latest versions from official RPi repos)

**Comment by HiassofT** [comment]
updated with noobs fixes from #3610 and bumped firmware to latest version as in #3605

**Comment by HiassofT** [comment]
I've added WIP label,

@MilhouseVH looked into kernel 5.1 options and this seems to be working well so it's best to head straight to current LE master kernel version without a detour to kernel 4.19.

The current kernel 5.1 WIP version of this PR is here: https://github.com/HiassofT/LibreELEC.tv/tree/le10-rpi4-kernel51, when we have an RPi kernel 5.1 with RPi4 support in master I'll switch this PR to a rebased version of that tree.

**Comment by HiassofT** [comment]
I've switched RPi4 to use kernel 5.1.15.

Note: this PR now depends on the linux-5.1.15 bump PR #3567

**Comment by HiassofT** [comment]
Rebased on top of LE master and runtime tested on RPi4.

This should be good to go in now.

**Pull request merged**